pkg/resource/logging-config/alloy-logging-config_test.go

```diff
@@ -0,0 +1,114 @@
+package loggingconfig
+
+import (
```

Comment:
I really hate those golden files :(

Reply:
I would prefer having 2 templates.based on the version

Reply:
Anyway, if I'm the only one that complaints then at least remove the update function. I want people to have to fix their template otherwise the golden files are used for PR reviews and not testing and they should not bé hère but in the PR description

Reply:
This part of the code becomes quite complex and we need a way to ensure we are not introducing unwanted changes. The update function makes it much easier to update tests, it should then be on the reviewer to asses that the changes are correct, we should not make our life harder by requiring to change those files manually.